awesome_sso/store/minio.py

```diff
@@ -58,51 +74,57 @@ def put(
         length: Optional[int] = None,
         content_type: str = "application/octet-stream",
     ):
+        """Uploads data from a stream to an object in a bucket."""
         if not length:
             length = len(data.read())
             data.seek(0)
-        self.client.put_object(
-            self.bucket, name, data, length, content_type=content_type
-        )
+
+        try:
+            self.client.put_object(
+                self.bucket, name, data, length, content_type=content_type
+            )
+        except S3Error as e:
+            logger.warning(e)
 
     def put_as_json(self, name: str, data: dict):
-        data_json = json.dumps(data)
-        data_bytes = data_json.encode("utf-8")
+        """Uploads data from a json to an object in a bucket."""
+        data_bytes = json.dumps(data).encode("utf-8")
         data_byte_stream = BytesIO(data_bytes)
-        length = len(data_byte_stream.read())
-        data_byte_stream.seek(0)
-        self.client.put_object(
-            self.bucket, name, data_byte_stream, length, content_type="application/json"
-        )
+
+        self.put(name, data_byte_stream, content_type="application/json")
 
     def get(self, name: str):
+        """Gets data of an object."""
         return self.client.get_object(self.bucket, name)
 
     def exists(self, name: str) -> bool:
+        """Check if object or bucket exist."""
         try:
             self.client.stat_object(self.bucket, name)
             return True
         except Exception:
             return False
 
     def remove_dir(self, folder: str):
+        """Remove folder."""
         self.logger.warning("removing %s", folder)
-        objects_to_delete = self.client.list_objects(
-            self.bucket, prefix=folder, recursive=True
-        )
-        object_names_to_delete = [x.object_name for x in objects_to_delete]
-        delete_objects = [DeleteObject(name) for name in object_names_to_delete]
-        self.logger.warning("Removing: %s", object_names_to_delete)
-        for del_err in self.client.remove_objects(self.bucket, delete_objects):
-            self.logger.warning("Deletion Error: %s", del_err)
+        objects_to_delete = self.list_objects(prefix=folder, recursive=True)
+
+        objects_to_delete += folder
+        self.logger.warning("Removing: %s", objects_to_delete)
+
+        self.remove_objects(objects_to_delete)
 
     def remove_object(self, name: str):
+        """Remove an object."""
         self.client.remove_object(self.bucket, name)
 
-    def upload_df(self, df: pd.DataFrame, name: str):
-        csv_bytes = df.to_csv(index=False).encode("utf-8")
-        csv_buffer = BytesIO(csv_bytes)
-        self.put(name, csv_buffer, content_type="application/csv")
+    def put_as_df(self, name: str, data: pd.DataFrame):
```

Comment:
plz change the name back so that we don't need to change all existing repository.